maps/torch/items/uniform_vendor.dm

```diff
@@ -36,4 +37,171 @@
 		ID = W
 		attack_hand(user)
 	else
-		..()
\ No newline at end of file
+		..()
+
+/obj/machinery/uniform_vendor/interface_interact(mob/user)
+	interact(user)
+	return TRUE
+
+/obj/machinery/uniform_vendor/interact(mob/user)
+	var/dat = list()
+	dat += "User ID: <a href='byond://?src=\ref[src];ID=1'>[ID ? "[ID.registered_name], [ID.military_rank], [ID.military_branch]" : "--------"]</a>"
+	dat += "<hr>"
+	if(!ID)
+		dat += "Insert your ID card to proceed."
+	else
+		uniforms = find_uniforms(ID.military_rank, ID.military_branch)
+		for(var/T in uniforms)
+			dat += "<b>[T]</b> <a href='byond://?src=\ref[src];get_all=[T]'>Select All</a>"
+			var/list/uniform = uniforms[T]
+			for(var/piece in uniform)
+				if(piece)
+					var/obj/item/clothing/C = piece
+					if(piece in selected_outfit)
+						dat += "<span class='linkOn'>[sanitize(initial(C.name))]</span><a href='byond://?src=\ref[src];rem=\ref[piece]'>X</a>"
+					else if (can_issue(C))
+						dat += "<a href='byond://?src=\ref[src];add=\ref[piece]'>[sanitize(initial(C.name))]</a>"
+					else
+						dat += "[sanitize(initial(C.name))] (ISSUED)"
+			dat += "<hr>"
+		dat += "<a href='byond://?src=\ref[src];vend=[1]'>Dispense</a>"
+	dat = jointext(dat,"<br>")
+	var/datum/browser/popup = new(user, "Uniform Dispenser", "Uniform Dispenser", 300, 700, src)
```

Comment:
could i ask you to convert this to nanoUI

Reply:
I have literally never worked with nanoui

Reply:
alright nvm